src/emulator/storage/apis/firebase.ts

```diff
@@ -128,6 +140,47 @@ export function createFirebaseEndpoints(emulator: StorageEmulator): Router {
     return res.json(new OutgoingFirebaseMetadata(metadata));
   });
 
+  firebaseStorageAPI.post(`/b/:bucketId/o/:objectId[(:)]generateSignedUrl`, async (req, res) => {
+    let signedUrlObject: SignedUrlResponse;
+
+    try {
+      signedUrlObject = await storageLayer.generateSignedUrl({
+        bucketId: req.params.bucketId,
+        decodedObjectId: decodeURIComponent(req.params.objectId),
+        authorization: req.header("authorization"),
+        originalUrl: `${req.protocol}://${req.hostname}:${req.socket.localPort}`,
+        ttlSeconds:
+          req.body.ttlSeconds === undefined ? SIGNED_URL_DEFAULT_TTL_SECONDS : req.body.ttlSeconds,
+      });
+    } catch (err) {
+      if (err instanceof NotFoundError) {
+        return res.status(404).json({
+          error: {
+            code: 404,
+            message: `Object in bucket does not exist`,
+          },
+        });
+      }
+      if (err instanceof ForbiddenError) {
+        return res.status(403).json({
+          error: {
+            code: 403,
+            message: `Permission denied. No WRITE permission.`,
```

Comment:
I'd leave the WRITE permission part out. Can you also send the message from `generateSignedUrl`? It helps people debugging if the error message appears where the error originated.

Reply:
Also, it's a `RulesetOperationMethod.GET` permission, isn't it?